### ISSUE TITLE
Allow multiple Proposals and/or one Commit in an MLSPlaintext

### ISSUE THREAD

**Issue by bifurcation** [mlswg/mls-protocol#100]
This allows combo operations -- multiple adds, moves, resyncs -- without having to worry about getting interrupted mid-operation.

**Comment by bifurcation** [comment]
Deferred for further discussion on the mailing list.

**Comment by bifurcation** [comment]
Repurposing this issue for combo moves with regard to Proposals and Commits.  We should cover the following cases:

* One or more Proposals with no Commit
* A Commit with no Proposals
* One or more Proposals bundled with a Commit

It seems like something like the following would be appropriate:

```
struct{
  Proposal proposals<0..2^16-1>;
  optional<Commit> commit;
} Handshake;
```

... together with a requirement that at least one of them be populated.

**Comment by bifurcation** [comment]
At the interim:
* Commit must cover all Proposals in the same message
* As a result, if any of the Proposals are invalid, then the whole thing fails

**Comment by bifurcation** [comment]
Fixed in #242
